Build failure: vfkit 0.6.0 on x86_64-darwin

## Steps to reproduce

The test fails.

I believe this started after https://github.com/NixOS/nixpkgs/pull/372672

This is used by podman

Steps to reproduce the behavior:

1. build *vfkit*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/sr4ci4jbqdn1c4l4b7jv42ivmdvqi1r1-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Building subPackage ./cmd/vfkit
# github.com/Code-Hex/vz/v3
In file included from _cgo_export.c:4:
In file included from socket.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/audio.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/bootloader.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/clipboard.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_13.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/configuration.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/console.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_13.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/entropy.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/graphics.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_13.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/keyboard.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/memory_balloon.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/network.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/osversion.go:6:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/platform.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/pointing_device.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/serial_console.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/shared_directory.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/socket.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/storage.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_11.m:7:
In file included from ./virtualization_11.h:9:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_12.m:7:
In file included from ./virtualization_12.h:7:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_12_3.m:7:
In file included from ./virtualization_12_3.h:9:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_13.m:7:
In file included from ./virtualization_13.h:9:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_14.m:7:
In file included from ./virtualization_14.h:13:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_15.m:4:
In file included from ./virtualization_15.h:10:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_debug.m:8:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_helper.m:7:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_view.m:7:
In file included from ./virtualization_view.h:9:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
buildPhase completed in 3 minutes 4 seconds
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
# github.com/Code-Hex/vz/v3
In file included from _cgo_export.c:4:
In file included from socket.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/audio.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/bootloader.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/clipboard.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_13.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/configuration.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/console.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_13.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/entropy.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/graphics.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_13.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/keyboard.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/memory_balloon.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/network.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/osversion.go:6:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/platform.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/pointing_device.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/serial_console.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/shared_directory.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/socket.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/storage.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization.go:6:
In file included from vendor/github.com/Code-Hex/vz/v3/virtualization_11.h:9:
vendor/github.com/Code-Hex/vz/v3/virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_11.m:7:
In file included from ./virtualization_11.h:9:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_12.m:7:
In file included from ./virtualization_12.h:7:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_12_3.m:7:
In file included from ./virtualization_12_3.h:9:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_13.m:7:
In file included from ./virtualization_13.h:9:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_14.m:7:
In file included from ./virtualization_14.h:13:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_15.m:4:
In file included from ./virtualization_15.h:10:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_debug.m:8:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_helper.m:7:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
# github.com/Code-Hex/vz/v3
In file included from virtualization_view.m:7:
In file included from ./virtualization_view.h:9:
./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
--- FAIL: TestStartIgnitionProvisionerServer (0.00s)
    main_test.go:37: 
        	Error Trace:	/private/tmp/nix-build-vfkit-0.6.0.drv-0/source/cmd/vfkit/main_test.go:37
        	Error:      	Received unexpected error:
        	            	Get "http://unix//virtiovsock": dial unix virtiovsock: connect: no such file or directory
        	Test:       	TestStartIgnitionProvisionerServer
FAIL
FAIL	github.com/crc-org/vfkit/cmd/vfkit	0.474s
FAIL
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-darwin"`
 - host os: `Darwin 24.2.0, macOS 15.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.25.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/v0g0bxsd5gw6k0jz2855f8h7l1218925-source`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@phaer @sarcasticadmin 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

hmmm, either I'm doing something wrong or this workaround does not work for me unfortunately.

I've run this:
```
nix build nixpkgs#legacyPackages.x86_64-darwin.vfkit --option sandbox true -L
```
(my nixpkgs are pinned to https://github.com/NixOS/nixpkgs/commit/5df43628fdf08d642be8ba5b3625a6c70731c19c)

and got this:
```
vfkit> In file included from ./virtualization_view.h:9:
vfkit> ./virtualization_helper.h:46:9: warning: macOS 15 API has been disabled [-W#pragma-messages]
vfkit> --- FAIL: TestStartIgnitionProvisionerServer (0.00s)
vfkit>     main_test.go:37:
vfkit>          Error Trace:    /private/tmp/nix-build-vfkit-0.6.0.drv-0/source/cmd/vfkit/main_test.go:37
vfkit>          Error:          Received unexpected error:
vfkit>                          Get "http://unix//virtiovsock": dial unix virtiovsock: connect: no such file or directory
vfkit>          Test:           TestStartIgnitionProvisionerServer
vfkit> FAIL
vfkit> FAIL     github.com/crc-org/vfkit/cmd/vfkit      0.459s
vfkit> FAIL
```

--- a/pkgs/by-name/vf/vfkit/package.nix
+++ b/pkgs/by-name/vf/vfkit/package.nix
@@ -4,6 +4,7 @@
   buildGoModule,
   darwin,
   fetchFromGitHub,
+  fetchpatch2,
   nix-update-script,
   testers,
   vfkit,
@@ -19,6 +20,13 @@ buildGoModule rec {
     rev = "v${version}";
     hash = "sha256-uBFL3iZLpGcVUSgZSoq8FI87CDAr3NI8uu+u5UsrZCc=";
   };
+
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/crc-org/vfkit/pull/276/commits/97ad151b208593d2fa0227ef5117f1ff2667b562.patch?full_index=1";
+      hash = "sha256-rcKLMXk5B2JoDpsNH4+aBmOidh9HtVQeYhx5q1raFJU=";
+    })
+  ];
 
   vendorHash = "sha256-+5QZcoI+/98hyd87NV6uX/VZqd5z38fk7K7RibX/9vw=";
 

